### PR TITLE
Add support for async/defer and appendTo props

### PR DESCRIPTION
**Pull request by dcop**
Added the support to load the script tag with `async` and `defer` attribute. 
Since I was at it I added the possibility to append the script to the body instead of the head element.

I was unable to test the `async` attribute since jsdom does not support it, but it seems quite trivial to conditionally add an attribute. I left the test as skipped, just in case, but if you want I can remove it.

### CODE REVIEW

File: src/google-recaptcha-provider.tsx [hunk 1]
```diff
@@ -131,6 +138,12 @@ export class GoogleReCaptchaProvider extends React.Component<
     if (!!nonce) {
       js.nonce = nonce;
     }
+    if (!!defer) {
```
Comment (t49tran) [inline src/google-recaptcha-provider.tsx:141]: I think it should be safe to do just:

```javascript
js.defer = !!defer;
```

and get rid of the `if` statement. Can you please try ?

File: __tests__/google-recaptcha-provider.test.tsx [hunk 2]
```diff
@@ -33,4 +33,36 @@ describe('<GoogleReCaptchaProvider />', () => {
...
+      <GoogleReCaptchaProvider reCaptchaKey="TESTKEY" nonce="NONCE" async>
+        <div />
+      </GoogleReCaptchaProvider>
+    );
+
+    const googleRecaptchaScript = (mountedComponent.instance() as GoogleReCaptchaProvider)
+      .generateGoogleReCaptchaScript();
+
+    expect(googleRecaptchaScript.outerHTML).toEqual(
+      '<script id="google-recaptcha-v3" ' +
```
Comment (t49tran) [inline __tests__/google-recaptcha-provider.test.tsx:64]: Can we please use ES6 template literals here instead of concat ?

**Review by t49tran — CHANGES_REQUESTED** [review]

**Comment by t49tran** [comment]
Hi @dcop , what you have done looks really good. I have requested some very minor changes and if you are happy to update it I can merge your PR in and do a release next week.

Thanks for contributing. It's really useful.

File: src/google-recaptcha-provider.tsx (hunk 1 above)
Comment (dcop) [reply src/google-recaptcha-provider.tsx]: Done!

File: __tests__/google-recaptcha-provider.test.tsx (hunk 2 above)
Comment (dcop) [reply __tests__/google-recaptcha-provider.test.tsx]: ...And done!

**Comment by dcop** [comment]
> Hi @dcop , what you have done looks really good. I have requested some very minor changes and if you are happy to update it I can merge your PR in and do a release next week.
> 
> Thanks for contributing. It's really useful.

No problem man, thank you for the effort of doing this. 
I've done what requested, should be ok now, if not I'll fix it.

**Review by t49tran — APPROVED** [review]

**Comment by alundiak** [comment]
I installed 1.6.1, and relied on README, but `appendTo` is not yet there. @dcop will you release release 1.6.2 soon?

PS. My error:

```
Type '{ children: Element; reCaptchaKey: string; appendTo: string; }' is not assignable to type 'IntrinsicAttributes & IntrinsicClassAttributes<GoogleReCaptchaProvider> & Readonly<PropsWithChildren<IGoogleReCaptchaProviderProps>>'.
  Property 'appendTo' does not exist on type 'IntrinsicAttributes & IntrinsicClassAttributes<GoogleReCaptchaProvider> & Readonly<PropsWithChildren<IGoogleReCaptchaProviderProps>>'.
```

**Pull request merged**